runtime/src/main/java/org/corfudb/protocols/wireprotocol/LogData.java

```diff
@@ -265,6 +265,7 @@ public void doSerialize(ByteBuf buf) {
     }
 
     void doSerializeInternal(ByteBuf buf) {
+        int startAddress = buf.readerIndex();
```

Comment:
should use buf.writerIndex()

Reply:
if we see the implementation of buf.readableBytes, it uses writerIndex-readerIndex, which is what I have in the function.  Let me know if I have missed something.